bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapse.jsx

```diff
@@ -21,13 +21,20 @@ const StyledLayerCollapsePanel = styled(LayerCollapsePanel)`
 `;
 
 const LayerCollapse = ({ groups, openGroupTitles, selectedLayerIds, controller }) => {
+
+    const [groupsData, setGroupsData] = useState([]);
+
+    useEffect(() => {
+        setGroupsData(groups);
+    },[groups]);
```

Comment:
I think the proper way to do this is with React.memo() and this feels very hackish :) Did you try if it makes a difference?

Reply:
I mean if it doesn't we can remove it. But if it does then I'm not sure if it's good for merging or if it should be refactored. I'll probably try removing it in local testing but I'm curious if you tested if this is something that is required for performance.

Reply:
I did not try react memo yet. It would probably be more suited for this one indeed. useState and useEffect can be removed from here.

Reply:
https://github.com/mr-lev1/oskari-frontend/pull/10

Reply:
Changes merged!